content/blog/2021-11-10-agent.md

```diff
@@ -0,0 +1,156 @@
+
+### Agent Mode Was Proven on Scale
+
+The next release of Prometheus will include Agent mode as an experimental feature. Flags, APIs and WAL format on disk might change. But the performance of the implementation is already battle-tested thanks to [Grafana Labs](https://grafana.com/) open-source work.
+
+Initial implementation of our Agent, custom WAL was inspired by current Prometheus TSDB WAL and created by [Robert Fratto](https://github.com/rfratto) in 2019, under the mentorship of [Tom Wilkie](https://twitter.com/tom_wilkie), Prometheus maintainer. It was then used in an open-source [Grafana Agent](https://github.com/grafana/agent) project that was since then, used by many Grafana Cloud customers and community members. Given the maturity of the solution, it was time to donate the implementation to Prometheus for native integration and bigger adoption. Robert (Grafana Labs), with the help of Srikrishna (Red Hat) and the community, ported the code to Prometheus codebase, which was merged to `main` 2 weeks ago! 🤗
+
+The donation process was quite smooth. Since some Prometheus maintainers saw and contributed to this code before on Grafana Agent, plus new WAL is inspired by the Prometheus server one, it was not hard for current Prometheus TSDB maintainers to take it under full maintenance! It's especially easier with Robert joining Prometheus Team as the TSDB maintainer (congratulations!).
+
+Now, let's explain how you can use it! (:
+
+### How to use Agent in details
+
+From now on, if print help of Prometheus, you should see more or less the following:
+
+```bash
+usage: prometheus [<flags>]
+
+The Prometheus monitoring server
+
+Flags:
+  -h, --help                     Show context-sensitive help (also try --help-long and --help-man).
+      (... other flags)
+      --storage.tsdb.path="data/"
+                                 Base path for metrics storage. Use with server mode only.
+      --storage.agent.path="data-agent/"
+                                 Base path for metrics storage. Use with agent mode only.
+      (... other flags)
+      --enable-feature= ...      Comma separated feature names to enable. Valid options: agent, exemplar-storage, expand-external-labels, memory-snapshot-on-shutdown, promql-at-modifier, promql-negative-offset, remote-write-receiver,
+                                 extra-scrape-metrics, new-service-discovery-manager. See https://prometheus.io/docs/prometheus/latest/feature_flags/ for more details.
+```
+
+Since agent mode is behind a feature flag, as mentioned previously, use `--enable-feature=agent` flag to run Prometheus in the Agent mode. Now, the rest of the flags are either for both server and Agent or only for a specific mode. You can see which flag is for what mode by checking the last sentence of flag help. `Use with server mode only` means it's only for server mode. If you don't see any mention like this, it means the flag is shared.
```

Comment:
```suggestion
Since the Agent mode is behind a feature flag, as mentioned previously, use the `--enable-feature=agent` flag to run Prometheus in the Agent mode. Now, the rest of the flags are either for both server and Agent or only for a specific mode. You can see which flag is for which mode by checking the last sentence of a flag's help string. "Use with server mode only" means it's only for server mode. If you don't see any mention like this, it means the flag is shared.
```